src/redis-cli.c

```diff
@@ -9184,18 +9287,41 @@ static void findBigKeys(int memkeys, unsigned memkeys_samples) {
                     exit(1);
                 }
 
-                printf(
-                   "[%05.2f%%] Biggest %-6s found so far '%s' with %llu %s\n",
-                   pct, type->name, type->biggest_key, sizes[i],
-                   !memkeys? type->sizeunit: "bytes");
+                /* We only show the original progress output when writing to a file */
+                if (!(isatty(STDOUT_FILENO) || getenv("FAKETTY"))) {
+                    printf("[%05.2f%%] Biggest %-6s found so far %s with %llu %s\n",
+                        pct, type->name, type->biggest_key, sizes[i],
+                        !memkeys? type->sizeunit: "bytes");
+
+                    /* Update overall progress */
+                    if(sampled % 1000000 == 0) {
+                        printf("[%05.2f%%] Sampled %llu keys so far\n", pct, sampled);
+                    }
+                }
 
                 /* Keep track of the biggest size for this type */
                 type->biggest = sizes[i];
             }
 
-            /* Update overall progress */
-            if(sampled % 1000000 == 0) {
-                printf("[%05.2f%%] Sampled %llu keys so far\n", pct, sampled);
+            /* Show the progress bar in TTY */
+            if (scan_loops % 10 == 0 && (isatty(STDOUT_FILENO) || getenv("FAKETTY"))) {            
```

Comment:
You are right about the "flickering". I agree that 10 is a bit too low. 100 is looking good on a local database, but when I tested over the network, I remember I waited a while before seeing the first progress information. I thought about a spinning cursor, or anything that gave a visual feedback to show the user we are not frozen. Let me do more test and see if 100 is ok most of the times.

Reply:
I created a database in the Cloud to test how long it will take to see the first refresh.
With 10, the first data is showing around 6 seconds.
With 100, it took around 90 seconds (I was expecting 60 seconds...).
Even with 10 it is a long time.
Would it be acceptable to introduce a refresh display based on time and not scan_loops? For instance every 300ms?
If so, I can give it a try.

Reply:
yeah, it is reasonable.

Reply:
The refresh is now every 300ms (`findBigKeys()`, `keyStats()`, and `findHotKeys()`).